tiledb/sm/rest/rest_client.h

```diff
@@ -190,6 +190,13 @@ class RestClient {
   /** Collection of extra headers that are attached to REST requests. */
   std::unordered_map<std::string, std::string> extra_headers_;
 
+  /** Collection of response headers that are attached to REST response. */
+  std::pair<std::string, std::unordered_map<std::string, std::string>>
```

Comment:
I understand the motivation for using a `pair` because it simplifies the logic in `Curl`, but this is the wrong approach: the `Rest` class does need to care how `Curl` is implemented. Only us an `std::unordered_map<std::string, std::string>` as the datatype.